app/models/manageiq/providers/autosde/storage_manager/event_catcher/stream.rb

```diff
@@ -0,0 +1,29 @@
+class ManageIQ::Providers::Autosde::StorageManager::EventCatcher::Stream
+  attr_reader :ems, :stop_polling, :poll_sleep
+
+  def initialize(options = {})
+    @ems = ManageIQ::Providers::Autosde::StorageManager.first
```

Comment:
This won't work if there is more than one instance of an Autosde provider.  Recommend passing in the ems as an arg:
```suggestion
  def initialize(ems, options = {})
    @ems = ems
```

Then when you initialize the Stream class in your Runner simply pass in the ems ivar:
`@event_monitor_handle ||= self.class.parent::Stream.new(@ems)`